hadoop-hdds/server-scm/src/main/java/org/apache/hadoop/hdds/scm/protocol/StorageContainerLocationProtocolServerSideTranslatorPB.java

```diff
@@ -329,6 +331,9 @@ public ScmContainerLocationResponse processRequest(
             "Unknown command type: " + request.getCmdType());
       }
     } catch (IOException e) {
+      if (SCMHAUtils.isRetriableWithNoFailoverException(e)) {
```

Comment:
Do we need this `RetriableWithNoFailoverException` ? 
Say a `ServiceException` containing `ReconfigurationInProgressException` should return `true` by `isRetriableWithNoFailoverException`.

Reply:
Given client side has already done the check, the server side can just return a `ServiceException` containing the real exception.

Reply:
> Do we want to address the suggested leader in a new Jira?

Yes, it will be addressed in a separate jira.

Reply:
> Given client side has already done the check, the server side can just return a `ServiceException` containing the real exception.

The check is required because for only certain exceptions , we throw ServiceException. For others, it sets an error code instead of throwing a service exception.

Reply:
> Given client side has already done the check, the server side can just return a `ServiceException` containing the real exception.

The Service exception is not thrown for all exceptions back from server. Only for certain exceptions, it throws service exception. Only StorageContainerLocation server seems to be sending back service exception back to client for each failure. I have remove the check in StorageContainerLocationServer.